File: The Bit Game.java
class Solution {
    public static int swapBitGame(long N) {
        // code here
        boolean player = false;
        StringBuilder sb = new StringBuilder(Long.toBinaryString(N));
        int strLen = sb.length(); 
        int counter = 0;
        while(counter < strLen){
            if(sb.charAt(counter) == '1'){
                player = !player;
            }
            counter++;
        }
        return player == false ? 2 : 1;
    }
}

File: The bit Game.cpp
class Solution{
public:
    int swapBitGame(long long N){
     int cnt=0;
     while(N>0){
         if((N&1)==1){
             cnt++;
         }
         N>>=1;
     }
     return cnt%2==0?2:1;
     }
};
